ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -521,17 +538,101 @@ def _run(self) -> Tuple[List[EventOddsRatio], bool]:
+                ),
+            }
+            return f"/api/person/funnel/correlation/?{urllib.parse.urlencode(params)}"
+
+        if self._filter.correlation_type == FunnelCorrelationType.PROPERTIES:
+            # NOTE: for property correlations, we just use the regular funnel
+            # persons endpoint, with the breakdown value set, and we assume that
+            # event.event will be of the format "{property_name}::{property_value}"
+            # NOTE: this is just trying to recreate what we are doing in the
+            # frontend here:
+            # https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelPropertyCorrelationTable.tsx#L77:L77
+            #
+            # That's a pretty complicated code path so I'd have low confidence
+            # that this was right.
+            property_name, property_value = event_definition["event"].split("::")
+            params = {
+                key: value
+                for key, value in self._filter.to_dict().items()
+                # NOTE: we want to filter anything about correlation, as the
+                # funnel persons endpoint does not understand or need these
+                # params.
+                if not key.startswith("funnel_correlation_")
+            }
+            params = {
+                **params,
+                "events": json.dumps(params["events"]),
+                "breakdown": property_name,
+                "funnel_step_breakdown": property_value,
+                "breakdown_type": "person",
+                "funnel_steps": len(params["events"]) if success else -2,
+                "display": "FunnelViz",
+                "funnel_custom_steps": json.dumps(
+                    [len(params["events"])] if success else list(range(1, len(params["events"])))
+                ),
+                "name": property_name,
+            }
+            return f"/api/person/funnel/?{urllib.parse.urlencode(params)}"
+
+        else:
+            return None
+
     def format_results(self, results: Tuple[List[EventOddsRatio], bool]) -> FunnelCorrelationResponse:
         return {
             "events": [
                 {
                     "success_count": odds_ratio["success_count"],
+                    "success_people_url": self.construct_people_url(success=True, event_definition=event_definition),
                     "failure_count": odds_ratio["failure_count"],
+                    "failure_people_url": self.construct_people_url(success=False, event_definition=event_definition),
                     "odds_ratio": odds_ratio["odds_ratio"],
                     "correlation_type": odds_ratio["correlation_type"],
-                    "event": self.serialize_event_with_property(odds_ratio["event"]),
+                    "event": event_definition,
                 }
                 for odds_ratio in results[0]
+                for event_definition in [self.serialize_event_with_property(odds_ratio["event"])]
```

Comment:
Q: Why do it like this? O.o

Reply:
This seems like a hack to get temporary variable assignment inside a list comprehension. It's too clever - we should do the simple stupid thing instead and extract `def format_event(odds_ratio)`

Reply:
Hah yes I couldn't quite figure out how to use create a var here. You can do `if (correlation := self.serialize_event_with_property(odds_ratio["event"]))` but that's also too clever and fails lint. I was hoping there would be a nice way to do this, but I'll replace with something boring